src/yadg/extractors/eclab/mpr.py

```diff
@@ -563,17 +565,55 @@ def process_modules(contents: bytes) -> tuple[dict, list, list, dict, dict]:
     return settings, params, ds, log, loop
 
 
+@singledispatch
+def extract_source(source, timezone):
```

Comment:
I think this should be just `extract(source: Any, timezone: str, **kwargs)` decorated by the `path -> source` or `fn -> source` deprecation warning.